.github/workflows/publish-release.yml

```diff
@@ -23,17 +23,15 @@ jobs:
           node-version-file: '.nvmrc'
       - name: Install Yarn
         run: corepack enable
-      - name: Restore Yarn cache
-        uses: actions/setup-node@v4
-        with:
-          node-version-file: '.nvmrc'
-          cache: yarn
-      - uses: actions/cache@v4
+      - name: Upload build artifacts
+        uses: actions/upload-artifact@v4
         with:
+          name: build-artifacts
+          retention-days: 4
+          include-hidden-files: true
```

Comment:
Added artifact handling to securely share files between jobs. This uploads:
- Built packages `(./packages/**/dist)`
- Yarn state file `(./node_modules/.yarn-state.yml)`
- Keeps artifacts for 4 days
- Includes hidden files which is needed for Yarn